database/entity.go

```diff
@@ -0,0 +1,87 @@
+package database
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+type Entity struct {
+	Id   int64
+	Name string
+}
+
+func (d *Database) CreateEntity(entity *Entity) error {
+	query := "INSERT INTO entities (name) VALUES (?) RETURNING id"
+	row := d.QueryRow(query, entity.Name)
+	return row.Scan(&entity.Id)
+}
+
+func (d *Database) GetEntity(id int64) (*Entity, error) {
+	d.lock.RLock()
+	defer d.lock.RUnlock()
+	query := "SELECT * FROM entities WHERE id = ?"
+	rows, err := d.Query(query, id)
```

Comment:
Why not `QueryRow`? We only need one result anyways

Reply:
because `parseEntity` requires wants a `sql.Rows` type